openpype/hosts/houdini/api/lib.py

```diff
@@ -989,22 +1001,30 @@ def self_publish():
     if result:
         return
 
-    current_node = hou.node(".")
-    inputs_paths = find_rop_input_dependencies(
-        current_node.inputDependencies()
-    )
-    inputs_paths.append(current_node.path())
+    if not nodes:
+        nodes = [hou.node(".")]
+
+    inputs_paths = [node.path() for node in nodes]
 
     host = registered_host()
-    context = CreateContext(host, reset=True)
+    context = CreateContext(host)
 
-    for instance in context.instances:
-        node_path = instance.data.get("instance_node")
-        instance["active"] = node_path and node_path in inputs_paths
+    if mode == 0:
+        for instance in context.instances:
+            node_path = instance.data.get("instance_node")
+            instance["active"] = node_path and node_path in inputs_paths
 
-    context.save_changes()
+        context.save_changes()
```

Comment:
Should this still validate whether any instances remained active just to ensure we're not running a redundant context save + publish?

Reply:
I think both of these should be outside the for loop. 
and it works the same as before. 
```
context.save_changes()
publisher_show_and_publish(comment)
```

Reply:
Well - in that case you don't need the `mode` argument at all.